scripts/create-conda-recipe.py

```diff
@@ -0,0 +1,188 @@
+import re
+from collections import defaultdict
+from copy import deepcopy
+
+import requests
+import yaml
+from bs4 import BeautifulSoup
+
+
+def get_extra_requires(path):
+    try:
+        with open(path) as fp:
+            extra_deps = defaultdict(set)
+            for k in fp:
+                if k.strip() and not k.startswith('#'):
+                    tags = set()
+                    if ':' in k:
+                        k, v = k.split(':')
+                        tags.update(vv.strip() for vv in v.split(','))
+
+                    k = re.split('([<>=])', k, 1)
+
+                    # If we have some version requrement, it will be returned as a
+                    # separate item by the split function
+                    if len(k) == 3:
+                        k = [k[0], k[1] + k[2]]
+                    assert len(k) <= 2, 'requirement should not have more than 2 parts'
+
+                    # Can not use extra requirements in conda - i.e. uvicorn[standard]
+                    k[0] = re.sub(r'\[\w+\]', '', k[0])
+
+                    # The docker package is docker-py on conda-forge
+                    if k[0] == 'docker':
+                        k[0] = 'docker-py'
+
+                    # Pytorch package is pytorch-cpu (gpu not needed for demo) on forge
+                    if k[0] == 'torch':
+                        k[0] = 'pytorch-cpu'
+
+                    # In conda recipe pkg name and version must be separated by space
+                    k = ' '.join(k)
+
+                    for t in tags:
+                        extra_deps[t].add(k)
+
+        return extra_deps
+    except FileNotFoundError:
+        return {}
+
+
+class RecipeDumper(yaml.SafeDumper):
+    """Adds a line break between top level objects and ignore aliases"""
+
+    def write_line_break(self, data=None):
+        super().write_line_break(data)
+        if len(self.indents) == 1:
+            super().write_line_break()
+
+    def ignore_aliases(self, data):
+        return True
+
+    def increase_indent(self, flow=False, *args, **kwargs):
+        return super().increase_indent(flow=flow, indentless=False)
+
+
+#######################################################
+# Get requirements from the extra-requirements.txt file
+#######################################################
+
+extra_deps = get_extra_requires('extra-requirements.txt')
```

Comment:
Is it possible to wrap each block into a function so that each function can be tested?